aggregator/src/aggregator/queue.rs

```diff
@@ -436,30 +438,27 @@ mod tests {
 
         let mut requests = Vec::new();
 
-        let backoff = ExponentialBackoff {
-            initial_interval: Duration::from_nanos(1),
-            max_interval: Duration::from_nanos(30),
-            multiplier: 2.0,
-            ..Default::default()
-        };
+        let backoff = ExponentialBuilder::new()
+            .with_min_delay(Duration::from_nanos(1))
+            .with_factor(2.0)
+            .with_max_times(30);
 
         debug!("spawning requests");
         for _ in 0..(concurrency as usize + depth) {
             let handler = Arc::clone(&handler);
-            let backoff = backoff.clone();
+            let backoff = backoff.build();
             requests.push(tokio::spawn({
                 async move {
-                    retry(backoff, || {
+                    (|| async {
                         let handler = Arc::clone(&handler);
```

Comment:
I think that since you remove the inner `async move` closure, we can remove this `clone`.